File: iris-ld/iris-ld-jsonld/src/main/java/city/smartb/iris/jsonld/jackson/JsonFieldReaderJackson.java
package city.smartb.iris.jsonld.jackson;

import city.smartb.iris.jsonld.reader.JsonField;
import city.smartb.iris.jsonld.reader.JsonFieldReader;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.util.Map;

public class JsonFieldReaderJackson implements JsonFieldReader {

     private final ObjectMapper mapper;

     public JsonFieldReaderJackson() {
          this(new ObjectMapperFactory().create());
     }

     public JsonFieldReaderJackson(ObjectMapper mapper) {
          this.mapper = mapper;
     }

     public JsonField read(Map<String, Object> json, String key) {
          return new JsonFieldJackson(mapper, json, key);
     }
}

File: iris-ld/iris-ld-jsonld/src/main/kotlin/city/smartb/iris/ld/jsonld/jackson/JsonFieldReaderJackson.kt
package city.smartb.iris.ld.jsonld.jackson

import city.smartb.iris.ld.jsonld.reader.JsonField
import city.smartb.iris.ld.jsonld.reader.JsonFieldReader
import com.fasterxml.jackson.databind.ObjectMapper

class JsonFieldReaderJackson @JvmOverloads constructor(
    private val mapper: ObjectMapper = ObjectMapperFactory().create()
) : JsonFieldReader {

    override fun read(json: Map<String, Any>, key: String): JsonField {
        return JsonFieldJackson(mapper, json, key)
    }

}
